### PR TITLE
improve aggregate performance by verifying the aggregate signature first

### PR DESCRIPTION
**Pull request by conradoplg**
Implements https://github.com/cfrg/draft-irtf-cfrg-frost/pull/360

### CODE REVIEW

File: frost-core/src/frost.rs [hunk 1]
```diff
@@ -396,8 +366,53 @@ where
...
         z,
-    })
+    };
+
+    // Verify the aggregate signature
+    let verification_result = pubkeys
+        .group_public
+        .verify(signing_package.message(), &signature);
+
+    // If verification failed; verify each share to find the cheater.
```
Comment (upbqdn) [inline frost-core/src/frost.rs:379]: The "only if" construct might be handy here:
```suggestion
    // Only if the verification of the aggregate signature failed; verify each share to find the cheater.
```

**Review by upbqdn — CHANGES_REQUESTED** [review]
PR looks great. I noticed an outdated comment that I think we should change before merging this PR, so I'm requesting changes.

See https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327

File: frost-core/src/frost.rs [hunk 2]
```diff
@@ -396,8 +366,53 @@ where
...
+    // This approach is more efficient since we don't need to verificate all shares
```
Comment (dconnolly) [inline frost-core/src/frost.rs:380]:
```suggestion
    // This approach is more efficient since we don't need to verify all shares
```

**Review by dconnolly — DISMISSED** [review]
🎉

**Review by dconnolly — APPROVED** [review]

**Review by upbqdn — COMMENTED** [review]
Looks great. However, I still think we should update this comment. https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327

Unfortunately, GitHub won't let me make a suggestion.

**Comment by conradoplg** [comment]
> Looks great. However, I still think we should update this comment.
> 
> https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327
> 
> Unfortunately, GitHub won't let me make a suggestion.

Good catch, fixed it in a904af6, I also removed some outdated wording

**Review by upbqdn — APPROVED** [review]

**Pull request merged**